Microsoft.CAP.Portal/Microsoft.CAP.Portal/Scripts/chartHelper.js

```diff
@@ -0,0 +1,233 @@
+﻿function createChartParams(chartTitle, results) {
+    //pre-processing
+    formatJsonDate(results);
+    getAnomalyPoints(results);
+    var showDate = getShowDate(results, 288);
+    var labelParams = getLabelParams(showDate.from,showDate.to);
+
+    var chartParams = {
+        title: {
+            text: chartTitle
+        },
+        legend: {
+            position: "bottom",
+            visible: true
+        },
+        chartArea: {
+            height: 512 //set height to be 512px
+        },
+        seriesDefaults: {
+            type: "line"
+        },
+        dataSource: results,
+        dateField: "Timestamp",
+        series: [{
+            name: "LowerBound", 
+            field: "LowerBound",
+            color: "gold",
+            zIndex: 1,
+            markers: {
+                visible: false
+            },
+            highlight: {
+                visible: false
+            },
+            tooltip: {
+                visible: false
+            }
+        }, {
+            name: "UpperBound", 
+            field: "UpperBound",
+            color: "gold",
+            zIndex: 1,
+            markers: {
+                visible: false
+            },
+            highlight: {
+                visible: false
+            },
+            tooltip: {
+                visible: false
+            }
+        }, {
+            name: "Value",  
+            field: "Value",
+            color: "deepskyblue",
+            zIndex: 2,
+            markers: {
+                size: 2
+            },
+            tooltip: {
+                visible: true,
+                opacity:0.8,
+                template: "Timestamp: #=dataItem.Timestamp # <br>\
+                               Value: #= dataItem.Value # <br>\
+                               IsAnomaly: #= dataItem.IsAnomaly # <br>\
+                               ExpectedValue: #= dataItem.ExpectedValue # <br>\
+                               LowerBound: #= dataItem.LowerBound # <br>\
+                               UpperBound: #= dataItem.UpperBound # <br>\
+                               ConfidenceIndex: #= dataItem.ConfidenceIndex #",
+            }
+        }, {
+            name: "Anomaly",  
+            field: "AnomalyValue",
+            color: "red",
+            opacity: 0, //hide the line
+            missingValues: "gap",   //not show the null points
+            zIndex: 3,  //set Anomaly to be the top layer
+            markers: {
+                size: 2,
+            },
+            tooltip: {
+                visible: false
+            }
+        }],
+        valueAxis: { 
+            labels: {
+                template: "#= value#"
+            }
+        },
+        categoryAxis: {
+            baseUnit: "minutes",
+            baseUnitStep: 5,//Take 5 minutes as the minimun interval at first
```

Comment:
You can pass TuningResponse.DataIntervalSeconds to get the accurate baseUnitStep for each data point